index.js

```diff
@@ -45,21 +56,18 @@ function isCurrentCityDataLoaded() {
   return cityByWeather.hasOwnProperty(currentCity);
 }
 
-renderApp();
-
 function reRenderApp() {
   // Need to split rendering cycles
-  setTimeout(() => window.renderApp(), 0);
+  setTimeout(window.renderApp, 0);
```

Comment:
Can you please describe why this is necessary?

When I wrote my version of the framework, I have used `setTimeout` to re-render the app after the event, just to give a time for event bubbling, etc.

This is a case? :)

Reply:
For now it's because of implementation - had to split the cycles because we assign all the content to the node, and changing vars in the process makes data shown outdated.
Should not be the case with promise-based data loading, will check that case.

Reply:
Can you please leave a comment? Cause this is a little bit unclear from the outside.

Reply:
No comments needed, promise-base API resolved this case. Thanks for the notice! 🙏